### ISSUE TITLE
[BUG] Default model evaluator "shap" is failing in explaining the model

### ISSUE THREAD

**Issue by bharadwaj509** [mlflow/mlflow#11111]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

- Client: 1.x.y
- Tracking server: 1.x.y

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS Sonoma 14.3.1
- **Python version**: 3.10
- **yarn version, if running the dev UI**:

### Describe the problem

I am trying to run mlflow.evaluate on a simple logistic regression model.

```
mlflow.evaluate(
        artifacts_uri,
        test,
        targets="quality",
        model_type="regressor",
        evaluators=["default"]
    )
```

The model is there in the artifacts URI, which ran succesfully in the previous runs and I can see the URI as well.
The test data and targets are loaded fine.
For evaluators, I used "Shap" but it throws me the following error.

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/mbp16/Desktop/Projects/MLFlowProject/pythonProject/evaluate.py", line 154, in <module>
    mlflow.evaluate(
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/mlflow/models/evaluation/base.py", line 1880, in evaluate
    evaluate_result = _evaluate(
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/mlflow/models/evaluation/base.py", line 1120, in _evaluate
    eval_result = evaluator.evaluate(
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/mlflow/models/evaluation/default_evaluator.py", line 1837, in evaluate
    evaluation_result = self._evaluate(model, is_baseline_model=False)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/mlflow/models/evaluation/default_evaluator.py", line 1754, in _evaluate
    self._log_artifacts()
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/mlflow/models/evaluation/default_evaluator.py", line 1587, in _log_artifacts
    self._log_model_explainability()
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/mlflow/models/evaluation/default_evaluator.py", line 953, in _log_model_explainability
    self._log_image_artifact(
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/mlflow/models/evaluation/default_evaluator.py", line 748, in _log_image_artifact
    pyplot.close(pyplot.gcf())
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/pyplot.py", line 1000, in gcf
    return figure()
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/pyplot.py", line 934, in figure
    manager = new_figure_manager(
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/pyplot.py", line 465, in new_figure_manager
    return _get_backend_mod().new_figure_manager(*args, **kwargs)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 3396, in new_figure_manager
    return cls.new_figure_manager_given_figure(num, fig)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 3401, in new_figure_manager_given_figure
    return cls.FigureCanvas.new_manager(figure, num)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 1785, in new_manager
    return cls.manager_class.create_with_canvas(cls, figure, num)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 2680, in create_with_canvas
    return cls(canvas_class(figure), num)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/backends/backend_macosx.py", line 149, in __init__
    FigureManagerBase.__init__(self, canvas, num)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 2654, in __init__
    self.toolbar = self._toolbar2_class(self.canvas)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/backends/backend_macosx.py", line 120, in __init__
    NavigationToolbar2.__init__(self, canvas)
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 2844, in __init__
    self._nav_stack = cbook._Stack()
  File "/usr/local/anaconda3/envs/aimlstuff/lib/python3.10/site-packages/matplotlib/_api/__init__.py", line 217, in __getattr__
    raise AttributeError(
AttributeError: module 'matplotlib.cbook' has no attribute '_Stack'. Did you mean: 'Stack'?

```

I am not sure how to fix it. The installation of Shap is succesful though.

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
REPLACE_ME
```

### Stack trace

```
REPLACE_ME
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [X] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@bharadwaj509 What is your matplotlib version?

**Comment by harupy** [comment]
@bharadwaj509 I don't think this is an mlflow issue. Can you reinstall matplotlib?

```
root@e853dc9e7d44:/# python
Python 3.10.13 (main, Nov  1 2023, 10:42:40) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import matplotlib
>>> matplotlib.cbook._Stack
<class 'matplotlib.cbook._Stack'>
>>> matplotlib.__version__
'3.8.2'
```

**Comment by harupy** [comment]
@bharadwaj509 Can you run the following code? Does that reproduce the error?

```
>>> import matplotlib.pyplot
>>> matplotlib.pyplot.figure()
```

**Comment by harupy** [comment]
https://stackoverflow.com/questions/77457355/attributeerror-module-matplotlib-cbook-has-no-attribute-stack

**Comment by bharadwaj509** [comment]
> @bharadwaj509 I don't think this is an mlflow issue. Can you reinstall matplotlib? Feel free to reopen the issue if it doesn't solve the issue.
> 
> ```
> root@e853dc9e7d44:/# python
> Python 3.10.13 (main, Nov  1 2023, 10:42:40) [GCC 12.2.0] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> import matplotlib
> >>> matplotlib.cbook._Stack
> <class 'matplotlib.cbook._Stack'>
> >>> matplotlib.__version__
> '3.8.2'
> ```

Apolgies for not replying early. My matplotlib is 3.8.0.d I will upgrade now and will test it again.

**Comment by bharadwaj509** [comment]
> ```python
> import matplotlib.pyplot as plt
> import numpy as np
> 
> # Data for plotting
> t = np.arange(0.0, 2.0, 0.01)
> s = 1 + np.sin(2 * np.pi * t)
> 
> fig, ax = plt.subplots()
> ax.plot(t, s)
> 
> ax.set(xlabel='time (s)', ylabel='voltage (mV)',
>        title='About as simple as it gets, folks')
> ax.grid()
> 
> fig.savefig("test.png")
> plt.show()
> ```

Thanks a lot for the reply.  I did the matplotlib update to 3.8.2 and still get the same issue.

**Comment by bharadwaj509** [comment]
Solution: I tried MATPLOTLIB 3.8.0 and 3.8.2. Since there is no other higher version available at this point, I installed 3.7.4 and it is working.
